test/NuGet.Core.Tests/NuGet.Packaging.Test/DependeciesGroupsForEachTFMRuleTests.cs

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Frameworks;
+using NuGet.Packaging.Rules;
+using Xunit;
+using NuGet.RuntimeModel;
+using NuGet.Client;
+using NuGet.ContentModel;
+using NuGet.Packaging.Core;
+
+namespace NuGet.Packaging.Test
+{
+    public class DependeciesGroupsForEachTFMRuleTests
+    {
+        ManagedCodeConventions _managedCodeConventions = new ManagedCodeConventions(new RuntimeGraph());
+        ContentItemCollection _collection = new ContentItemCollection();
+
+        [Fact]
+        public void Validate_PackageWithDependenciesForEachTFMInLib_ShouldNotWarn()
+        {
+            var frameworks = new string[]
+            {
+                ".NETFramework2.0",
+                ".NETFramework3.5",
+                ".NETFramework4.0",
+                ".NETFramework4.5",
+                ".NETStandard1.0",
+                ".NETStandard1.3",
+                ".NETStandard2.0"
+            }.Select(f => NuGetFramework.Parse(f));
```

Comment:
This is a neat idea, I like it. I don't know if it's better or worse, but you could consider

```csharp
var frameworks = new NuGetFramework[]
{
   FrameworkConstants.CommonFrameworks.NetFramework20,
   FrameworkConstants.CommonFrameworks.NetFramework35,
   FrameworkConstants.CommonFrameworks.NetFramework40,
   FrameworkConstants.CommonFrameworks.NetFramework45,
   FrameworkConstants.CommonFrameworks.NetStandard10,
   FrameworkConstants.CommonFrameworks.NetStandard13,
   FrameworkConstants.CommonFrameworks.NetStandard20,
}
```
This way less parsing needs to happen, particularly if this pattern is used by all the tests.